server/api/index.js

```diff
@@ -0,0 +1,64 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const Recipe = require("./models/recipe");
+
+const { ApolloServer, gql } = require("apollo-server-express");
+
+const typeDefs = gql`
+  input RecipeInput {
+    servings: Int,
+    prepTime: String,
+    cookTime: String,
+    ingredients: [IngredientInput],
+    steps: [String],
+    links: [String],
+    name: String
+  }
+  input IngredientInput {
+    measurement: String!,
+    quantity: Int!,
+    name: String!
+  }
+  type Ingredient {
+    measurement: String!,
+    quantity: Int!,
+    name: String!
+  }
+  type Recipe {
+    servings: Int!,
+    prepTime: String,
+    cookTime: String,
+    ingredients: [Ingredient!],
+    steps: [String!],
+    links: [String],
+    name: String!
+  }
+  type Query { recipes: [Recipe] }
+  type Mutation { createRecipe(input: RecipeInput): Recipe }
+`;
+
+const resolvers = {
+    Query: { 
+      recipes: async() => {
+        return await Recipe.find({});
+      } 
+    },
+    Mutation: {
+      createRecipe: async(parent, args) => {
+        return await Recipe.create(args.input);
+      }
+    }
+};
+
+mongoose.connect("mongodb://server:42pickleheads@ds363118.mlab.com:63118/pickleheads-cookbook")
+
+const app = express();
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  introspection: true,
+  playground: true,
+});
+server.applyMiddleware({ app, path: "/", cors: true });
```

Comment:
We may want to change this so that the overall endpoint is `/api/graphql`
```suggestion
server.applyMiddleware({ app, path: "/graphql", cors: true });
```

Reply:
Yeah I back

Reply:
This didn't work, gave me a 404 with `/api/graphql`

Reply:
Huh, I wonder why

Reply:
It's probably because on now it deploy everything at api/{filename} so if you change the filename to graphql it should work. So when lucas is hitting api/graphql it's looking for a file graphql.{ext} which doesn't exist so 404

Reply:
Yeah that worked